File: C++/Disk.cpp
#include <iostream>
#include <string>

using namespace std;

class Disk
{
    private:
        string type;        // menampung tipe
        string capacity;    // menampung kapasitas
        int price;          // menampung harga
    
    public:
        // constructor
        Disk() {}

        Disk(string type, string capacity, int price)
        {
            this->type = type;
            this->capacity = capacity;
            this->price = price;
        }

        // setter and getter
        void setType(string type)
        {
            this->type = type;
        }

        string getType()
        {
            return this->type;
        }

        void setCapacity(string capacity)
        {
            this->capacity = capacity;
        }

        string getCapacity()
        {
            return this->capacity;
        }

        void setPrice(int price)
        {
            this->price = price;
        }

        int getPrice()
        {
            return this->price;
        }

        // method
        void printDisk()
        {
            cout << "Disk Type          : " << this->getType() << endl;
            cout << "Disk Capacity      : " << this->getCapacity() << endl;
            cout << "Disk Price         : Rp " << this->getPrice() << ",-" << endl;
        }

        // destructor
        ~Disk() {}
};

File: Java/Disk.java
public class Disk {
    private String type;        // menampung tipe
    private String capacity;    // menampung kapasitas
    private int price;          // menampung harga

    // constructor
    public Disk() {
    }

    public Disk(String type, String capacity, int price) {
        this.type = type;
        this.capacity = capacity;
        this.price = price;
    }

    // setter and getter
    public void setType(String type) {
        this.type = type;
    }

    public String getType() {
        return this.type;
    }

    public void setCapacity(String capacity) {
        this.capacity = capacity;
    }

    public String getCapacity() {
        return this.capacity;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public int getPrice() {
        return this.price;
    }

    // method
    public void printDisk() {
        System.out.println("Disk Type           : " + this.getType());
        System.out.println("Disk Capacity       : " + this.getCapacity());
        System.out.println("Disk Price          : Rp " + this.getPrice() + ",-");
    }
}
